src/import-utils.ts

```diff
@@ -73,97 +73,88 @@ export async function inlineHtmlImport(
+      astUtils.removeElementAndNewline(linkTag);
       return;
     }
 
     const relative =
-        urlUtils.relativeUrl(docUrl, importBundleUrl) || importBundleUrl;
-    dom5.setAttribute(htmlImport, 'href', relative);
+        urlUtils.relativeUrl(document.url, importBundleUrl) || importBundleUrl;
+    dom5.setAttribute(linkTag, 'href', relative);
     visitedUrls.add(importBundleUrl);
     return;
   }
 
-  let importSource: string;
-  try {
-    importSource = await loader(resolvedImportUrl);
-  } catch (e) {
-    throw new Error(`Unable to load ${resolvedImportUrl}: ${e.message}`);
+  // If the analyzer could not load the import document, we can't inline it, so
+  // lets skip it.
+  const htmlImport = findInSet(
+      document.getByKind('html-import', {imported: true}),
+      (i) => i.url === resolvedImportUrl);
+  if (!htmlImport) {
+    return;
   }
 
-  const importDoc = parse5.parseFragment(importSource);
-  rewriteDocumentToEmulateBaseTag(resolvedImportUrl, importDoc);
-  rewriteDocumentBaseUrl(importDoc, resolvedImportUrl, docUrl);
-  const nestedImports = dom5.queryAll(importDoc, matchers.htmlImport);
+  // When inlining html documents, we'll parse it as a fragment so that we do
+  // not get html, head or body wrappers.
+  const importAst =
+      parse5.parseFragment(htmlImport.document.parsedDocument.contents);
+  rewriteAstToEmulateBaseTag(importAst, resolvedImportUrl);
+  rewriteAstBaseUrl(importAst, resolvedImportUrl, document.url);
+  const nestedImports = dom5.queryAll(importAst, matchers.htmlImport);
 
   // Move all of the import doc content after the html import.
-  astUtils.insertAllBefore(
-      htmlImport.parentNode!, htmlImport, importDoc.childNodes!);
-  astUtils.removeElementAndNewline(htmlImport);
+  astUtils.insertAllBefore(linkTag.parentNode!, linkTag, importAst.childNodes!);
+  astUtils.removeElementAndNewline(linkTag);
 
   // Recursively process the nested imports.
   for (const nestedImport of nestedImports) {
     await inlineHtmlImport(
-        docUrl, nestedImport, visitedUrls, docBundle, manifest, loader);
+        document, nestedImport, visitedUrls, docBundle, manifest);
   }
 }
 
 /**
  * Inlines the contents of the document returned by the script tag's src url
  * into the script tag content and removes the src attribute.
  */
-export async function inlineScript(
-    docUrl: UrlString,
-    externalScript: ASTNode,
-    loader: (url: UrlString) => Promise<string>) {
-  const rawUrl = dom5.getAttribute(externalScript, 'src')!;
-  const resolvedUrl = urlLib.resolve(docUrl, rawUrl);
-  let script: string|undefined = undefined;
-  try {
-    script = await loader(resolvedUrl);
-  } catch (e) {
-    // If a script doesn't load, skip inlining.
-    // TODO(usergenic): Add plylog logging for load error.
-    console.warn(`Unable to load script at ${resolvedUrl}: ${e.message}`);
-  }
-
-  if (script === undefined) {
+export async function inlineScript(document: Document, scriptTag: ASTNode) {
```

Comment:
This seems like it'll work as long as the document already had the imported URL in its transitive imports. Otherwise, getByKind won't return anything. You might need to pass in a collection of documents?

I'd make sure you have a test for this scenario.